### PR TITLE
Tests in extensions have been updated to use JUnit5 annotations

### LINKED ISSUE #2849: Use JUnit5 extensions instead of deprecated Junit4 Rules / Remove `EnableRuleMigrationSupport`
Right now in `kubernetes-tests/` module, we're using `@EnableRuleMigrationSupport` annotation to allow usage of Junit4 rules and we're using `KubernetesServer` `@Rule` for Kubernetes Mock Server that is used in each test:
https://github.com/fabric8io/kubernetes-client/blob/33a735a61a121083e4dfd0e9ba9105e62936cca6/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ConfigMapTest.java#L34-L38

Due to this, we're also dependent on `junit-jupiter-migrationsupport`:

https://github.com/fabric8io/kubernetes-client/blob/33a735a61a121083e4dfd0e9ba9105e62936cca6/kubernetes-tests/pom.xml#L47-L51

I added Junit5 extensions for Kubernetes Mock Server ([EnableKubernetesMockClient](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/EnableKubernetesMockClient.java#L35) for Kubernetes and [EnableOpenShiftMockClient](https://github.com/fabric8io/kubernetes-client/blob/master/openshift-server-mock/src/main/java/io/fabric8/openshift/client/server/mock/EnableOpenShiftMockClient.java#L35)) in https://github.com/fabric8io/kubernetes-client/pull/2785 which can be used to initialize both `KubernetesMockServer` and `KubernetesClient`. So one can easily refactor the above test to use Junit5 extension annotation like this:
```java
@EnableKubernetesMockClient
class ConfigMapTest {
  KubernetesMockServer server; // initialized via above annotation
  KubernetesClient client;     // initialized via above annotation

  @Test
  void testLiteralConfigMap() throws InterruptedException {
     // ...
  }

  @Test
  void testFromResourceWithFileConfigMap() throws InterruptedException {
     // ...
  }

  @Test
  void testFromResourceConfigMap() throws InterruptedException {
     // ...
  }
}
```

In case anyone is interested, it would be nice if we could move from Junit4 Rules to new Junit5 extension annotations and get rid of this dependency as well. You can easily find occurrences of `EnableRuleMigrationSupport` annotation with a simple grep like this:
```
git grep -c "EnableRuleMigrationSupport"
```

### PR DESCRIPTION
**Pull request by MUzairS15**
## Description

Fix #2849: Add JUnit5 Annotations for extension mockservers
    
Updated all tests in extensions to use Junit5 implementations, also updated POM files.

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [x] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
Need to be squashed when we'll merge.

**Comment by rohanKanojia** [comment]
Now that extensions are using JUnit5 annotations, please remove this dependency from all extension poms. You can find all occurrences of this dependency with this simple command:
```
git grep -in junit-jupiter-migrationsupport
```

**Review by shawkins — APPROVED** [review]
LGTM

File: extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1CertificateCrudTest.java [hunk 1]
```diff
@@ -19,33 +19,31 @@
...
 
-@EnableRuleMigrationSupport
+@EnableCertManagerMockClient(crud=true)
 class V1CertificateCrudTest {
-
-  @Rule
-  public CertManagerServer server = new CertManagerServer(true, true);
+    CertManagerClient client;
+//  @Rule
+//  public CertManagerServer server = new CertManagerServer(true, true);
```
Comment (rohanKanojia) [inline extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1CertificateCrudTest.java:34]: Please remove this commented out code

File: extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1alpha2CertificateCrudTest.java [hunk 2]
```diff
@@ -19,33 +19,31 @@
...
 
-@EnableRuleMigrationSupport
+@EnableCertManagerMockClient(crud = true)
 class V1alpha2CertificateCrudTest {
 
-  @Rule
-  public CertManagerServer server = new CertManagerServer(true, true);
-
+//  @Rule
+//  public CertManagerServer server = new CertManagerServer(true, true);
```
Comment (rohanKanojia) [inline extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1alpha2CertificateCrudTest.java:34]: Please remove this commented out code.

File: extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1alpha3CertificateCrudTest.java [hunk 3]
```diff
@@ -19,33 +19,31 @@
...
 
-@EnableRuleMigrationSupport
+@EnableCertManagerMockClient(crud = true)
 class V1alpha3CertificateCrudTest {
 
-  @Rule
-  public CertManagerServer server = new CertManagerServer(true, true);
-
+//  @Rule
+//  public CertManagerServer server = new CertManagerServer(true, true);
```
Comment (rohanKanojia) [inline extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1alpha3CertificateCrudTest.java:34]: Please remove this

File: extensions/chaosmesh/tests/src/test/java/io/fabric8/chaosmesh/test/crud/IoChaosTest.java [hunk 4]
```diff
@@ -16,29 +16,22 @@
...
-import io.fabric8.chaosmesh.server.mock.ChaosMeshServer;
-import org.junit.Rule;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
 
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (rohanKanojia) [inline extensions/chaosmesh/tests/src/test/java/io/fabric8/chaosmesh/test/crud/IoChaosTest.java:26]: Could you please avoid using wildcard imports here?

**Review by rohanKanojia — COMMENTED** [review]
looks good with minor comments. Please avoid using wildcard imports(maybe it's not configured correctly in your IDE) and remove commented out lines.

File: extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1alpha3CertificateCrudTest.java (hunk 3 above)
Comment (MUzairS15) [reply extensions/certmanager/tests/src/test/java/io/fabric8/certmanager/test/crud/V1alpha3CertificateCrudTest.java]: DONE

File: extensions/service-catalog/tests/src/test/java/io/fabric8/servicecatalog/test/crud/ServiceInstanceTest.java [hunk 5]
```diff
@@ -19,26 +19,19 @@
...
+import static org.junit.jupiter.api.Assertions.*;
 
-@EnableRuleMigrationSupport
+@EnableServiceCatalogMockClient(crud=true)
 class ServiceInstanceTest {
 
-    @Rule
-    public ServiceCatalogServer server = new ServiceCatalogServer(true, true);
-
+//
```
Comment (rohanKanojia) [inline extensions/service-catalog/tests/src/test/java/io/fabric8/servicecatalog/test/crud/ServiceInstanceTest.java:30]: typo?

Comment (MUzairS15) [reply extensions/service-catalog/tests/src/test/java/io/fabric8/servicecatalog/test/crud/ServiceInstanceTest.java]: yes,Thankyou

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**